ypathgen/pathgen_test.go

```diff
@@ -1757,13 +1861,15 @@ func TestGenerateDirectorySnippet(t *testing.T) {
 		inDirectory               *ygen.Directory
 		inListBuilderKeyThreshold uint
 		inPathStructSuffix        string
-		want                      GoPathStructCodeSnippet
-		wantNoWildcard            GoPathStructCodeSnippet
+		inSplitByModule           bool
+		want                      []GoPathStructCodeSnippet
+		wantNoWildcard            []GoPathStructCodeSnippet
 	}{{
 		name:        "container-with-config",
 		inDirectory: directories["/root-module/container-with-config"],
-		want: GoPathStructCodeSnippet{
+		want: []GoPathStructCodeSnippet{{
 			PathStructName: "ContainerWithConfig",
+			Package:        "device",
```

Comment:
Can you also test with a different package name? I suspect there is some handling required in generator.go to write the fakeroot's package differently from the other packages. Maybe you can fallthrough to the next case.

Reply:
well that logic was missing, added